src/main/scala/dsptools/numbers/chisel_types/SIntTypeClass.scala

```diff
@@ -0,0 +1,141 @@
+// See LICENSE for license details.
+
+package dsptools.numbers
+
+import chisel3._
+import chisel3.util.{ShiftRegister, Cat}
+import dsptools.{hasContext, DspContext, Grow, Wrap, Saturate, DspException}
+import chisel3.experimental.FixedPoint
+
+import scala.language.implicitConversions
+
+/**
+  * Defines basic math functions for SInt
+  */
+trait SIntRing extends Any with Ring[SInt] with hasContext {
+  def zero: SInt = 0.S
+  def one: SInt = 1.S
+  def plus(f: SInt, g: SInt): SInt = {
+    // TODO: Saturating mux should be outside of ShiftRegister
+    val sum = {
+      if(context.overflowType == Grow) f +& g
```

Comment:
Saw this in the previous round of reviews and didn't think of it, but this should probably be a match statement instead of if else if ... else